src/main/java/de/epiceric/shopchest/utils/ShopUtils.java

```diff
@@ -18,6 +18,7 @@
 public class ShopUtils {
 
     private final HashMap<Location, Shop> shopLocation = new HashMap<>();
+    private final Collection<Shop> shopLocationValues = Collections.unmodifiableCollection(shopLocation.values());
```

Comment:
Yes, `shopLocationValues` is a view of hashmap values, and passing it into `unmodifiableCollection` prevent add/remove operations.

I tested it myself to be sure:
```java
Map<Integer, Integer> map = new HashMap<>();
map.put(1, 101);
map.put(2, 102);

Collection<Integer> values = Collections.unmodifiableCollection(map.values());
System.out.println("size " + values.size());

map.put(3, 103);
map.put(4, 104);

try {
    System.out.println("edit collection " + values.remove(102));
} catch (Exception e) {
    System.out.println("cant edit");
}

System.out.println("size " + values.size());
```
Did output:
```
size 2
cant edit
size 4
```